### DISCUSSION TITLE
No way to test guest syscall code

### DISCUSSION THREAD

**Discussion by lrettig** [athenavm/athena#289] [Ideas]
Right now this code is cfg-gated, both in the SDK:

https://github.com/athenavm/athena/blob/461e631660e869768c5b042563e227e2d09159e8/vm/sdk/src/lib.rs#L6-L18

as well as in the underlying syscode implementation:

https://github.com/athenavm/athena/blob/461e631660e869768c5b042563e227e2d09159e8/vm/entrypoint/src/syscalls/host.rs#L18-L42

The former won't even compile for the host; the latter will compile but can't be executed.

@poszu any ideas how we can allow testing of this code?

Related: #109, #107

**Comment by poszu** [comment]
I'm unsure, they would need to be executed in a VM. However, this code is just a few lines of raw assembly. I think we can get away with reviewing these functions "with eyes" and depending on all other tests to execute them for now.
